pkg/universe.dagger.io/x/benjamin.reigner@epitech.eu/gcp/function/function.cue

```diff
@@ -0,0 +1,49 @@
+package function
+
+import (
+	"dagger.io/dagger"
+	"universe.dagger.io/docker"
+	"universe.dagger.io/x/benjamin.reigner@epitech.eu/gcp/gcr"
+)
+
+// The runtimes are copied from https://cloud.google.com/functions/docs/concepts/exec
+// If the list would come to change please submit an issue or a pull request to integrate it
+#Runtime: "nodejs16" | "nodejs14" | "nodejs12" | "nodejs10" | "nodejs8" | "nodejs6" | "python39" | "python38" | "python37" | "go116" | "go113" | "go111" | "java11" | "dotnet3" | "ruby27" | "ruby26" | "php74"
+
+// The definition used to deploy a serverless cloud function
+#Function: {
+
+	// The Config from gcpServerless/configServerless.#Config
+	config: gcr.#Credentials
```

Comment:
`config`'s purpose is to mount the secret token in the container.
This secret can be retrieve thanks `gcr.#Credentials` but it should be possible to set it from a secret environment or a command executed in the client.

That should be a `dagger.#Secret` and you should create a simple `#Image` definition that contains everything required to execute `gcloud` configuration or let user provide his own one.